beacon.go

```diff
@@ -17,110 +17,164 @@
 package beacon
 
 import (
+	"context"
 	"encoding/xml"
 	"errors"
-	"golang.org/x/oauth2/google"
-	"google.golang.org/api/genomics/v1"
-	"google.golang.org/appengine"
+	"fmt"
 	"net/http"
+	"os"
 	"strconv"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/appengine"
 )
 
 type beaconConfig struct {
-	variantSetIds []string
+	projectID string
+	tableId   string
 }
 
+const (
+	projectKey = "GOOGLE_CLOUD_PROJECT"
+	bqTableKey = "GOOGLE_BIGQUERY_TABLE"
+)
+
 var config = beaconConfig{
-	variantSetIds: []string{"3049512673186936334"},
+	projectID: os.Getenv(projectKey),
+	tableId:   os.Getenv(bqTableKey),
 }
 
 func init() {
 	http.HandleFunc("/", handler)
 }
 
-func requestToSearch(r *http.Request) (*genomics.SearchVariantsRequest, string, error) {
-	allele := r.FormValue("allele")
-	search := &genomics.SearchVariantsRequest{
-		ReferenceName: r.FormValue("chromosome"),
-		VariantSetIds: config.variantSetIds,
+func handler(w http.ResponseWriter, r *http.Request) {
```

Comment:
*We can ignore this for this change, and do it in a follow up*

It's nice to keep the HTTP machinery away from the logic.  See how it's done with the api package in the htsget server.  This allows you to have regular functions (that take regular arguments and return regular values and errors).

The original code here was closer to this ideal.  Let's get this in and then we can go through how to refactor.

Reply:
done